lib/game.rb

```diff
@@ -1,41 +1,51 @@
 require_relative "display"
 require_relative "messages"
-require_relative "rules"
 require_relative "player"
+require_relative "human_player"
+require_relative "computer_player"
+require_relative "rules"
+require_relative "move_validator"
+require_relative "set_up"
 
 class Game
 
-  def initialize(player_one, player_two, display, messages)
-    @player_one = player_one
-    @player_two = player_two
+  def initialize(display, rules, set_up)
     @display = display
-    @messages = messages
+    @rules = rules
+    @set_up = set_up
   end
 
-  def run
-    welcome_users
-    @player_one.choose
-    @player_two.choose
-    determine_winner
-    present_winner
+  def play
+    @display.welcome_users
+    @display.choose_player
+    player = @set_up.players(@display.sets_up_game)
+    player_turn(player[0])
+    player_turn(player[1])
+    result(player[0], player[1])
   end
 
-  def welcome_users
-    @display.present(@messages.welcome_message)
+  def player_turn(player)
+    @display.get_name(player.position)
```

Comment:
Should getting a player name be part of the player turn? I think it could be nice to ask both players for their names before either have their turns. This will separate responsibility between methods.

Reply:
Also, I feel that setting players up with names should be part of the set up of a game.